hicommonwealth/commonwealth
# Archive backend

## Link to Issue
Closes: #4190
Closes: #4467

## Description of Changes
- Adds route to archive a thread
- Adds route to unarchive a thread
- Updates the `createThreadComment` and `createThreadReaction` methods so that they fail if the thread is archived

## Test Plan
- Find a thread, get the ID
- Make an authenticated admin request to `POST http://localhost:8080/api/threads/<ThreadID>/archive` – confirm that the payload shows `archived_at` with the current timestamp
- Make an authenticated admin request to `POST http://localhost:8080/api/threads/<ThreadID>/unarchive` – confirm that the payload shows `archived_at` with a value of `null`

Unit tests are included to ensure comment/reaction creation is blocked for archived threads.

## Deployment Plan
N/A

## Other Considerations
Users can still update/delete their comments and react on existing comments.
